platforms/core-runtime/launcher/src/main/java/org/gradle/tooling/internal/provider/ProviderConnection.java

```diff
@@ -346,10 +346,23 @@ private Parameters initParams(ProviderOperationParameters operationParameters) {
             daemonParams.setBaseDir(operationParameters.getDaemonBaseDir());
         }
 
-        //override the params with the explicit settings provided by the tooling api
-        List<String> jvmArguments = operationParameters.getJvmArguments();
-        if (jvmArguments != null) {
-            daemonParams.setJvmArgs(jvmArguments);
+        // Since 8.13, we split the daemon JVM args into base and additional JVM args (#31462)
+        // When calling the following new methods, the following can happen:
+        // - < 8.13: the new split will return a null
+        // - >=8.13: the new split will return a non-null value, clearly signifying that the method is supported.
+        List<String> baseJvmArguments = operationParameters.getBaseJvmArguments(null);
```

Comment:
Idea: You can create a local (empty) list, pass it as arguments and check for reference equality to see if `getBaseJvmArguments` was available on the `operationParameters` proxy.